engine/runtime/src/main/java/org/enso/compiler/TreeToIr.java

```diff
@@ -126,16 +127,18 @@ IR.Module translateModule(Tree module) {
               var t = translateModuleSymbol(def);
               bindings = cons(t, bindings);
             }
-            case Tree.ArgumentBlockApplication app -> {
-              var t = translateComment(app);
-              bindings = cons(t, bindings);
-            }
             case Tree.Annotated anno -> {
               var n = new IR$Name$Annotation("@" + anno.getAnnotation().codeRepr(), getIdentifiedLocation(anno), meta(), diag());
               bindings = cons(n, bindings);
               var t = translateModuleSymbol(anno.getExpression());
               bindings = cons(t, bindings);
             }
+            case Tree.Documented doc -> {
+              var c = translateComment(doc);
+              bindings = cons(c, bindings);
+              var t = translateModuleSymbol(doc.getExpression());
+              bindings = cons(t, bindings);
+            }
```

Comment:
What if something is both annotated and documented? I'd use recursion to handle node types that act as wrappers in any order.

Reply:
Made more recursive in [0086305](https://github.com/enso-org/enso/pull/3806/commits/008630563777d284f8eeedae3fa7b7c7ee7751cb)